fabric-tool-attribute-api-v1/src/main/java/net/fabricmc/fabric/impl/tool/attribute/handlers/TaggedToolsModdedBlocksToolHandler.java

```diff
@@ -47,7 +47,7 @@ public ActionResult isEffectiveOn(Tag<Item> tag, BlockState state, ItemStack sta
 			if (entry != null) {
 				int miningLevel = 0; // minimum mining level: the tool is tagged but nothing else is said about it
 				int requiredMiningLevel = entry.getMiningLevel(tag);
-				return requiredMiningLevel >= miningLevel ? ActionResult.SUCCESS : ActionResult.PASS;
+				return requiredMiningLevel >= 0 && miningLevel >= requiredMiningLevel ? ActionResult.SUCCESS : ActionResult.PASS;
```

Comment:
honestly this could probably be simplified to requiredMiningLevel == 0, not sure if it would make it less clear however

Reply:
Yeah I know, but my formatting preserves the "intention" behind the code. I wonder if the compiler knows how to optimise it...

I guess it could always be `== 0` with the logic explained in a comment if needed?